doc/developer-guides/hld/hv-ioc-virt.rst

```diff
@@ -0,0 +1,726 @@
+| supporting HW reset can  | challenging to manage    |                          |
+| be passed through to a   | devices on UOS crashes   |                          |
+| UOS.                     |                          |                          |
++--------------------------+--------------------------+--------------------------+
+
+
+Interface Specification
+***********************
+
+Doxygen-style comments in the code are used for interface specification.
+This section provides some examples on how functions and structure
+should be commented.
+
+Function Header Template
+========================
+
+.. code-block:: c
+
+   /**
+    * @brief Initialize environment for Trusty-OS on a VCPU.
+    *
+    * More info here.
+    *
+    * @param[in]    vcpu Pointer to VCPU data structure
+    * @param[inout] param guest physical address. This gpa points to
+    *                     struct trusty_boot_param
+    *
+    * @return 0 - on success.
+    * @return -EIO - (description when this error can happen)
+    * @return -EINVAL - (description )
+    *
+    * @pre vcpu must not be NULL.
+    * @pre param must ...
+    *
+    * @post the return value is non-zero if param is ....
+    * @post 
+    *
+    * @remark The api must be invoked with interrupt disabled.
+    * @remark (Other usage constraints here)
+    */
+
+
+Structure
+=========
+
+.. code-block:: c
+
+   /**
+    * @brief An mmio request.
+    *
+    * More info here.
+    */
+   struct mmio_request {
+           uint32_t direction;  /**< Direction of this request. */
+           uint32_t reserved;   /**< Reserved. */
+           int64_t address;     /**< gpa of the register to be accessed. */
+           int64_t size;        /**< Width of the register to be accessed. */
+           int64_t value;       /**< Value read from or to be written to the
+   register. */
+   } __aligned(8);
+
+
+IOC Mediator Configuration
+**************************
+
+IOC Mediator Usage
+******************
+
+The device model configuration command syntax for IOC mediator is as
+follows::
+
+   -i,[ioc_channel_path],[wakeup_reason]
+   -l,[lpc_port],[ioc_channel_path]
+
+The "ioc_channel_path" is an absolute path for communication between
+IOC mediator and UART DM.
+
+The "lpc_port" is "com1" or "com2", IOC mediator needs one unassigned
+lpc port for data transfer between UOS and SOS.
+
+The "wakeup_reason" is IOC mediator boot up reason, each bit represents
+one wakeup reason.
+
+For example, the following commands are used to enable IOC feature, the
+initial wakeup reason is the ignition button and cbc_attach uses ttyS1
+for TTY line discipline in UOS::
+
+   -i /run/acrn/ioc_$vm_name,0x20
+   -l com2,/run/acrn/ioc_$vm_name
+
+
+Porting and adaptation to different platforms
+*********************************************
+
+Reference Documents
+*******************
+
+- 519325_CBC_Protocol_Service_Interfaces_Specification_Rev1.2.pdf
```

Comment:
@liuyuan-GitHub Can you provide links to these (public) documents

Reply:
These reference documents are in my local, do we have a public shared directory for these documents?

Reply:
Please remove reference documents